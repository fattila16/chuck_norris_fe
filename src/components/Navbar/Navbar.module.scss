@import 'src/assets/scss/vars';
@import 'src/assets/scss/color-palette';

.nav {
    display: flex;
    justify-content: center;
    align-items: center;
    height: $navbar-height;
    position: fixed;
    z-index: 100000;
    top:0;
    left: 0;
    width: 100%;
    padding: 0 10px;
    background: map-get($active-theme, primary-color);
}