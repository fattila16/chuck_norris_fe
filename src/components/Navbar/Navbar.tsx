import React, { useCallback, ChangeEvent, useState } from 'react';
import LinkList, { NavLink } from './LinkList/LinkList';
import css from './Navbar.module.scss';
import NAVBAR_LINKS from './navbar.const';
import ToggleSwitch from '../ToggleSwitch/ToggleSwitch';
import useRTL, { Direction } from '../../hooks/useRTL';
export interface NavbarProps {
    logo: string;
    links: NavLink[];
}
const Navbar: React.FC = () => {
    const { links } = NAVBAR_LINKS;
    const { setDirection } = useRTL();
    const [isChecked, setIsChecked] = useState(false);
    const handleToggleSwitch = useCallback(
        (event: ChangeEvent<HTMLInputElement>) => {
            const { checked } = event.target;
            console.log(checked);
            if (checked) setDirection(Direction.RTL);
            else setDirection(Direction.LTR);
            setIsChecked(!isChecked);
        },
        [isChecked, setDirection],
    );
    return (
        <>
            <nav className={css.nav}>
                <section>
                    <LinkList links={links} />
                </section>
                <section>
                    <ToggleSwitch onChange={handleToggleSwitch} checked={isChecked} />
                </section>
            </nav>
        </>
    );
};

export default Navbar;
