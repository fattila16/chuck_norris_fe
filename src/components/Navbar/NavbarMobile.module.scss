@import 'src/assets/scss/vars';
@import 'src/assets/scss/color-palette';

.nav {
    display: flex;
    flex-direction: column;
    position: fixed;
    z-index: 100000;
    top: 0;
    left: 0;
    width: 100%;
    background: map-get($active-theme, primary-color);

    section {
        display: flex;
        padding: 0 10px;
    }

    .header {
        justify-content: space-between;
        height: 100%;
        align-items: center;
        height: $navbar-height-mobile;
    }

    .content {
        display: none;
    }

    &.open {
        .content {
            display: flex;
            justify-content: center;
            flex-direction: column;
            align-items: center;
        }
    }
}