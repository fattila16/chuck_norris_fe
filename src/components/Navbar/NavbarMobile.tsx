import React, { useCallback, useState, ChangeEvent } from 'react';
import Hamburger from '../Hamburger/Hamburger';
import ToggleSwitch from '../ToggleSwitch/ToggleSwitch';
import LinkList from './LinkList/LinkList';
import useRTL, { Direction } from '../../hooks/useRTL';
import css from './NavbarMobile.module.scss';
import Router from 'next/router';
import NAVBAR_LINKS from './navbar.const';

const NavbarMobile: React.FC = () => {
    const [open, setOpen] = useState(false);
    Router.events.on('routeChangeStart', () => setOpen(false));
    const { links } = NAVBAR_LINKS;
    const { setDirection } = useRTL();
    const [isChecked, setIsChecked] = useState(false);
    const handleToggleSwitch = useCallback(
        (event: ChangeEvent<HTMLInputElement>) => {
            const { checked } = event.target;
            if (checked) setDirection(Direction.RTL);
            else setDirection(Direction.LTR);
            setIsChecked(!isChecked);
        },
        [isChecked, setDirection],
    );
    return (
        <nav className={`${css.nav} ${open ? css.open : ''}`}>
            <section className={css.header}>
                <Hamburger open={open} setOpen={setOpen} />
            </section>
            <section className={css.content}>
                <LinkList links={links} />
                <span>
                    <ToggleSwitch onChange={handleToggleSwitch} checked={isChecked} textTrue="RTL" textFalse="LTR" />
                </span>
            </section>
        </nav>
    );
};

export default NavbarMobile;
