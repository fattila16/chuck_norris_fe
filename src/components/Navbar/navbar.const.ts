import { NavbarProps } from './NavbarDefault';

export const NAVBAR_LINKS: NavbarProps = {
    logo: '/',
    links: [
        {
            name: 'Categories',
            href: '/categories',
        },
        {
            name: 'Search',
            href: '/search',
        },
    ],
};

export default NAVBAR_LINKS;
