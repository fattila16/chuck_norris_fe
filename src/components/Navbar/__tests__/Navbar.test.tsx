import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '../Navbar';

describe('[Navbar]', () => {
    it('Should render Navbar component without error', () => {
        const component = shallow(<Navbar />);
        expect(component.getElements()).toMatchSnapshot();
    });
});
