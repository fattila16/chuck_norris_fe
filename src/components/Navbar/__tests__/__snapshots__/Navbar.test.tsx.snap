// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Navbar] Should render Navbar component without error 1`] = `
Array [
  <React.Fragment>
    <nav
      className="nav"
    >
      <section>
        <LinkList
          links={
            Array [
              Object {
                "href": "/categories",
                "name": "Categories",
              },
              Object {
                "href": "/search",
                "name": "Search",
              },
            ]
          }
        />
      </section>
      <section>
        <ToggleSwitch
          checked={false}
          onChange={[Function]}
        />
      </section>
    </nav>
  </React.Fragment>,
]
`;
